### PR TITLE
Auth inter check

### PR DESCRIPTION
**Pull request by MaGaMeGa**

### CODE REVIEW

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java [hunk 1]
```diff
@@ -437,7 +437,44 @@ public void removeInterCloudAuthorizationEntryById(final long id) {
 		}		
 	}
 	
-	
+	//-------------------------------------------------------------------------------------------------
+	public InterCloudAuthorizationCheckResponseDTO checkInterCloudAuthorizationResponse(long cloudId,
```
Comment (rbocsi) [inline authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: finals

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java [hunk 2]
```diff
@@ -437,7 +437,44 @@ public void removeInterCloudAuthorizationEntryById(final long id) {
...
+			long serviceDefinitionId) {
+		
+		logger.debug("checkInterCloudAuthorizationRequestResponse started...");
+		
+		try {
+			final boolean isCloudIdInvalid = cloudId < 1 || !cloudRepository.existsById(cloudId);
```
Comment (rbocsi) [inline authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: here just check <1

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java [hunk 3]
```diff
@@ -437,7 +437,44 @@ public void removeInterCloudAuthorizationEntryById(final long id) {
...
+		
+			if (isCloudIdInvalid || isServiceDefinitionIdInvalid) {
+				String exceptionMsg = "Following parameters are invalid:";
+				exceptionMsg = isCloudIdInvalid ? exceptionMsg + " 'cloud id' ," : exceptionMsg;
+				exceptionMsg = isServiceDefinitionIdInvalid ? exceptionMsg + " 'serviceDefinition id' ," : exceptionMsg;
+				exceptionMsg = exceptionMsg.substring(0, exceptionMsg.length() -1);
+				throw new InvalidParameterException(exceptionMsg);
+			}
+			
+			Cloud validCloud = cloudRepository.findById(cloudId).get();
```
Comment (rbocsi) [inline authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: check here the extistense

**Review by rbocsi — CHANGES_REQUESTED** [review]
see inline comments

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java (hunk 1 above)
Comment (MaGaMeGa) [reply authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: done

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java (hunk 2 above)
Comment (MaGaMeGa) [reply authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: done

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java (hunk 3 above)
Comment (MaGaMeGa) [reply authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: done

**Review by rbocsi — APPROVED** [review]
seems ok

**Pull request merged**